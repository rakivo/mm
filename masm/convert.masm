_start:
; f2i
    push 420.0
    f2i
    dmp 1

; f2u
    push 420.0
    f2u
    dmp 1

; i2f
    push 420
    i2f
    dmp 1

; i2u
    push 420
    i2u
    dmp 1

; u2i
    push 420
    u2i
    dmp 1

; u2f
    push 420
    u2f
    dmp 1

; these will fail:

    ; push -420
    ; i2u
    ; dmp 1

    ; push -420.0
    ; f2u
    ; dmp 1

    ret
