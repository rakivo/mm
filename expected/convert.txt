420
420
420.0
420
420
420.0
